src/components/LmsConfigurations/DegreedIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,356 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import {
+  ValidationFormGroup, Input, StatefulButton, Icon,
+} from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import LmsApiService from '../../data/services/LmsApiService';
+import StatusAlert from '../StatusAlert';
+import SUBMIT_STATES from '../../data/constants/formSubmissions';
+import { handleErrors, validateLmsConfigForm } from './common';
+
+export const REQUIRED_DEGREED_CONFIG_FIELDS = [
+  'degreedBaseUrl',
+  'degreedUserId',
+  'degreedUserPassword',
+  'secret',
+  'degreedCompanyId',
+  'key',
+];
+
+function configFormReducer(state, action) {
+  switch (action.type) {
+    case 'PENDING': {
+      return { ...state, error: null, submitState: SUBMIT_STATES.PENDING };
+    }
+    case 'ERROR': {
+      return { ...state, error: action.value, submitState: SUBMIT_STATES.ERROR };
+    }
+    case 'INVALID': {
+      return {
+        ...state,
+        submitState: SUBMIT_STATES.ERROR,
+        invalidFields: action.value,
+        error: 'Form could not be submitted as there are fields with invalid values. Please correct them below.',
+      };
+    }
+    case 'COMPLETE': {
+      return {
+        ...state,
+        invalidFields: {},
```

Comment:
Don't these items here match the default state already? Do you have to explicitly set them to invalid?

Reply:
If I follow correctly, it seems like the flow you'd expect would be starting with all required fields being invalid and then instead of adding to an object/array containing those fields, you would remove them when they are valid? Or am I completely misunderstanding?

In the current implementation we start with no field being invalid and iterate through the required fields and the fields in the form data to determine what is missing and error accordingly, so yes they have to be explicitly set as invalid. While typing this I was thinking through the reverse and I'm not sure that'd be a very easy proposition.